.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run tests
+      run: sbt clean coverage test coverageReport
+    - name: Coverage Aggregate
+      run: sbt coverageReport
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        fail_ci_if_error: true
```

Comment:
This doesn't cache.

Something like this (pulled from vinyldns) would include caching and easier to compile against multiple jdks...

```
# Much copied from sbt-github-actions, modified to support running e2e tests
name: Continuous Integration

on:
  pull_request:
    branches: ['*']
  push:
    branches: ['master']

env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

jobs:
  build:
    name: Build and Test
    if: "!contains(github.event.head_commit.message, 'ci skip')"
    strategy:
      matrix:
        os: [ubuntu-latest]
        scala: [2.12.10]
        java: [adopt@1.11]
    runs-on: ${{ matrix.os }}
    steps:
      - name: Checkout current branch (full)
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Setup Java and Scala
        uses: olafurpg/setup-scala@v5
        with:
          java-version: ${{ matrix.java }}

      - name: Cache ivy2
        uses: actions/cache@v1
        with:
          path: ~/.ivy2/cache
          key: ${{ runner.os }}-sbt-ivy-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (generic)
        uses: actions/cache@v1
        with:
          path: ~/.coursier/cache/v1
          key: ${{ runner.os }}-generic-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (linux)
        if: contains(runner.os, 'linux')
        uses: actions/cache@v1
        with:
          path: ~/.cache/coursier/v1
          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (macOS)
        if: contains(runner.os, 'macos')
        uses: actions/cache@v1
        with:
          path: ~/Library/Caches/Coursier/v1
          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (windows)
        if: contains(runner.os, 'windows')
        uses: actions/cache@v1
        with:
          path: ~/AppData/Local/Coursier/Cache/v1
          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache sbt
        uses: actions/cache@v1
        with:
          path: ~/.sbt
          key: ${{ runner.os }}-sbt-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - run: sbt ++${{ matrix.scala }} validate verify

      - name: Codecov
        uses: codecov/codecov-action@v1
        with:
          fail_ci_if_error: true
```

Reply:
Ah cool! Will refactor!